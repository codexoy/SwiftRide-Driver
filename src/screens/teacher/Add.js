import React from "react"
import { View, Text, Button, StyleSheet, ScrollView } from "react-native"
import { connect } from "react-redux"
import { strings } from "../../i18n"
import Notification from "../../components/Notification"
import PageTitle from "../../components/PageTitle"
import NotificationButtons from "./NotificationButtons"

class Add extends React.Component {
	static navigationOptions = () => {
		return {
			title: "add",
			tabBarLabel: strings("tabs.add"),
			tabBarAccessibilityLabel: strings("tabs.add"),
			tabBarTestID: "AddTab"
		}
	}
	render() {
		return <ScrollView style={styles.container} />
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1
	},
	notifications: {
		flex: 1,
		paddingRight: 30,
		paddingLeft: 30,
		marginTop: 20
	},
	notification: {}
})

export default connect()(Add)
