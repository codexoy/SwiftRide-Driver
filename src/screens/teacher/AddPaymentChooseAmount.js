import React from "react"
import {
	View,
	StyleSheet,
	Text,
	TouchableHighlight,
	TouchableOpacity,
	TextInput,
	KeyboardAvoidingView,
	Platform
} from "react-native"
import { connect } from "react-redux"
import { strings } from "../../i18n"
import { Icon } from "react-native-elements"
import PageTitle from "../../components/PageTitle"
import {
	MAIN_PADDING,
	floatButtonOnlyStyle,
	DEFAULT_MESSAGE_TIME,
	fullButton
} from "../../consts"
import { API_ERROR } from "../../reducers/consts"
import { NavigationActions } from "react-navigation"
import SlidingMessage from "../../components/SlidingMessage"
import { fetchOrError } from "../../actions/utils"
import { popLatestError } from "../../actions/utils"

export class AddPaymentChooseAmount extends React.Component {
	constructor(props) {
		super(props)
		this.state = {
			amount: "000",
			slidingMessageVisible: false
		}

		this.addPayment = this.addPayment.bind(this)
	}

	componentDidUpdate() {
		const error = this.props.dispatch(popLatestError(API_ERROR))
		if (error) {
			this.setState({
				error,
				slidingMessageVisible: true
			})
		}
	}

	addPayment = async () => {
		const resp = await this.props.dispatch(
			fetchOrError("/teacher/add_payment", {
				method: "POST",
				body: JSON.stringify({
					amount: parseInt(this.state.amount),
					student_id: this.props.navigation.getParam("student")
						.student_id
				})
			})
		)
		if (resp) {
			this.setState({ error: "", slidingMessageVisible: true }, () => {
				setTimeout(
					() =>
						this.props.navigation.dispatch(
							NavigationActions.back()
						),
					DEFAULT_MESSAGE_TIME
				)
			})
		}
	}

	changeAmount = amount => {
		this.setState({ amount })
	}

	render() {
		return (
			<View style={styles.container}>
				<SlidingMessage
					visible={this.state.slidingMessageVisible}
					error={this.state.error}
					success={strings("teacher.add_payment.success")}
					close={() =>
						this.setState({ slidingMessageVisible: false })
					}
				/>
				<View style={styles.headerRow}>
					<TouchableOpacity
						onPress={() => {
							this.props.navigation.goBack()
						}}
					>
						<Icon name="arrow-forward" type="material" />
					</TouchableOpacity>
					<PageTitle
						style={styles.title}
						title={strings("teacher.add_payment.title2")}
					/>
				</View>
				<KeyboardAvoidingView
					behavior={Platform.OS === "ios" ? "position" : "height"}
					keyboardVerticalOffset={Platform.select({
						ios: 62,
						android: 24
					})}
					style={styles.container}
				>
					<View style={styles.amountContainer}>
						<TextInput
							placeholder="000.00"
							value={this.state.amount}
							onChangeText={amount => this.changeAmount(amount)}
							style={styles.amountInput}
							autoFocus={true}
							maxLength={5}
							keyboardType="number-pad"
						/>
					</View>
					<TouchableHighlight
						underlayColor="#ffffff00"
						onPress={this.addPayment}
						style={styles.floatButton}
					>
						<Text style={styles.buttonText}>{strings("done")}</Text>
					</TouchableHighlight>
				</KeyboardAvoidingView>
			</View>
		)
	}
}

function mapStateToProps(state) {
	return {
		fetchService: state.fetchService,
		errors: state.errors
	}
}
export default connect(mapStateToProps)(AddPaymentChooseAmount)

const styles = StyleSheet.create({
	container: {
		flex: 1
	},
	title: {
		marginLeft: 8,
		marginTop: -4
	},
	headerRow: {
		flexDirection: "row",
		padding: 0,
		justifyContent: "flex-start",
		marginLeft: MAIN_PADDING,
		marginRight: MAIN_PADDING,
		marginTop: 20
	},
	amountInput: {
		fontSize: 80,
		alignItems: "center",
		fontFamily: "Assistant-Light"
	},
	amountContainer: {
		borderBottomWidth: 0,
		backgroundColor: "#f4f4f4",
		padding: 20,
		marginTop: 80
	},
	buttonText: {
		color: "#fff",
		fontWeight: "bold",
		fontSize: 20
	},
	floatButton: {
		...fullButton
	}
})
