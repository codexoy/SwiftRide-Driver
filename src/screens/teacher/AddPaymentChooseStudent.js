import React from "react"
import {
	View,
	StyleSheet,
	Text,
	FlatList,
	TouchableHighlight
} from "react-native"
import { connect } from "react-redux"
import { strings } from "../../i18n"
import Row from "../../components/Row"
import UserWithPic from "../../components/UserWithPic"
import Separator from "../../components/Separator"
import { SearchBar, Button, Icon } from "react-native-elements"
import PageTitle from "../../components/PageTitle"
import { MAIN_PADDING } from "../../consts"
import { getStudents } from "../../actions/students"
import { NavigationActions } from "react-navigation"
import EmptyState from "../../components/EmptyState"
import StudentsLoader from "../../components/StudentsLoader"

export class AddPaymentChooseStudent extends React.Component {
	constructor(props) {
		super(props)
		this.state = {
			search: "",
			student: {},
			students: [],
			page: 1,
			nextUrl: "",
			loading: true
		}

		this.renderItem = this.renderItem.bind(this)
		this._getStudents()
	}

	updateSearch = search => {
		this.setState({ search, loading: true }, () => {
			this._getStudents(false)
		})
	}

	_getStudents = async (append = true) => {
		resp = await getStudents(this.props.fetchService, this.state)
		let newValue = resp.students
		if (append) {
			newValue = [...this.state.students, ...newValue]
		}
		this.setState({
			students: newValue,
			nextUrl: resp.nextUrl,
			loading: false
		})
	}

	endReached = () => {
		if (!this.state.nextUrl) return
		this.setState(
			{
				page: this.state.page + 1
			},
			() => {
				this._getStudents()
			}
		)
	}

	onPress = item => {
		this.setState(
			{
				student: item
			},
			() => {
				this.props.navigation.navigate("Second", { student: item })
			}
		)
	}

	renderItem = ({ item, index }) => {
		let noneMargin = {}
		if (index == 0) {
			noneMargin = { marginTop: 0 }
		}
		return (
			<TouchableHighlight
				underlayColor="#f9f9f9"
				key={`student${item.student_id}`}
				onPress={() => this.onPress(item)}
			>
				<Row
					leftSide={
						<View style={styles.addButton}>
							<Icon name="ios-add" type="ionicon" color="#000" />
						</View>
					}
					style={{ ...styles.row, ...noneMargin }}
				>
					<UserWithPic
						name={item.user.name}
						nameStyle={styles.nameStyle}
						width={64}
						height={64}
					/>
				</Row>
			</TouchableHighlight>
		)
	}

	_renderEmpty = () => (
		<EmptyState
			image="students"
			text={strings("no_students")}
			style={styles.empty}
		/>
	)

	_renderStudents = () => {
		if (this.state.loading) {
			return (
				<View style={styles.listLoader}>
					<StudentsLoader />
				</View>
			)
		}
		return (
			<FlatList
				data={this.state.students}
				keyboardShouldPersistTaps="always"
				renderItem={this.renderItem}
				keyExtractor={item => `student${item.student_id}`}
				onEndReached={this.endReached}
				keyboardDismissMode="on-drag"
				keyboardShouldPersistTaps="always"
				ListEmptyComponent={this._renderEmpty}
			/>
		)
	}
	render() {
		return (
			<View style={styles.container}>
				<PageTitle
					style={styles.title}
					title={strings("teacher.add_payment.title1")}
					leftSide={
						<Button
							icon={
								<Icon
									name="ios-close"
									type="ionicon"
									size={36}
								/>
							}
							onPress={() => {
								this.props.navigation.dispatch(
									NavigationActions.back()
								)
							}}
							type="clear"
							style={styles.closeButton}
						/>
					}
				/>
				<View
					style={styles.studentsSearchView}
					testID="StudentsSearchView"
				>
					<SearchBar
						placeholder={strings("teacher.students.search")}
						onChangeText={this.updateSearch}
						value={this.state.search}
						platform="ios"
						containerStyle={styles.searchBarContainer}
						inputContainerStyle={styles.inputContainerStyle}
						cancelButtonTitle={strings("teacher.students.cancel")}
						inputStyle={styles.search}
						textAlign="right"
						autoFocus={true}
						testID="searchBar"
					/>
					<Separator />
					{this._renderStudents()}
				</View>
			</View>
		)
	}
}

function mapStateToProps(state) {
	return {
		fetchService: state.fetchService
	}
}

export default connect(mapStateToProps)(AddPaymentChooseStudent)

const styles = StyleSheet.create({
	container: {
		flex: 1,
		marginLeft: MAIN_PADDING,
		marginRight: MAIN_PADDING,
		marginTop: 20
	},
	studentsSearchView: { paddingTop: 0 },
	searchBarContainer: {
		backgroundColor: "transparent",
		paddingBottom: 0,
		paddingTop: 0,
		marginTop: 16
	},
	inputContainerStyle: {
		borderRadius: 30,
		paddingLeft: 8,
		paddingTop: 6,
		paddingBottom: 6,
		width: "100%",
		marginLeft: 0,
		marginRight: 0
	},
	search: {
		alignItems: "flex-start",
		paddingLeft: 6,
		fontSize: 14,
		marginLeft: 0
	},
	nameStyle: {
		fontSize: 18,
		marginTop: 14
	},
	addButton: {
		marginTop: 12
	},
	row: { marginTop: 20 },
	closeButton: {
		marginTop: -6
	}
})
