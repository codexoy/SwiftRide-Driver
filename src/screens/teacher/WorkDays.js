import React, { Fragment } from "react"
import {
	View,
	StyleSheet,
	Text,
	TouchableOpacity,
	Platform,
	ScrollView,
	TouchableHighlight,
	Alert
} from "react-native"
import { connect } from "react-redux"
import { strings, errors } from "../../i18n"
import PageTitle from "../../components/PageTitle"
import {
	MAIN_PADDING,
	colors,
	fullButton,
	API_DATE_FORMAT,
	SHORT_API_DATE_FORMAT
} from "../../consts"
import { Icon } from "react-native-elements"
import DateTimePicker from "react-native-modal-datetime-picker"
import moment from "moment"
import { API_ERROR } from "../../reducers/consts"
import { fetchOrError, popLatestError } from "../../actions/utils"

const DEFAULT_HOURS = {
	from_hour: 8,
	from_minutes: 0,
	to_hour: 17,
	to_minutes: 0
}
export class WorkDays extends React.Component {
	constructor(props) {
		super(props)
		this.state = {
			isTimePickerVisible: false,
			daysWithHours: {},
			fromSchedule: this.props.navigation.getParam("fromSchedule")
		}

		this.days = [0, 1, 2, 3, 4, 5, 6]
		if (this.state.fromSchedule) {
			// this means we need to add work hours
			// to specific dates
			this.days = this._getWeekDays()
		}
		this.days.forEach(day => {
			this.state.daysWithHours[day] = []
		})

		this._getDays()
	}

	_getDays = async () => {
		try {
			let extra
			if (this.state.fromSchedule) {
				const startOfWeek = moment()
					.startOf("week")
					.format(SHORT_API_DATE_FORMAT)
				const endOfWeek = moment()
					.endOf("week")
					.format(SHORT_API_DATE_FORMAT)
				extra =
					"?on_date=ge:" + startOfWeek + "&on_date=le:" + endOfWeek
			}
			const resp = await this.props.fetchService.fetch(
				"/teacher/work_days" + extra,
				{
					method: "GET"
				}
			)
			let day,
				newState = { ...this.state.daysWithHours }
			resp.json["data"].forEach(hour => {
				if (!hour["on_date"] && this.state.fromSchedule) {
					// when we're in a specific week, we are only interested in specific work days
					return
				}
				day = hour["day"]
				if (hour["on_date"]) {
					day = moment
						.utc(hour["on_date"])
						.format(SHORT_API_DATE_FORMAT)
				}
				newState[day].push(hour)
			})
			this.setState({
				daysWithHours: newState
			})
		} catch (error) {}
	}

	_getWeekDays = () => {
		var date = moment().startOf("week"),
			weeklength = 7,
			result = []
		while (weeklength--) {
			result.push(date.format(SHORT_API_DATE_FORMAT))
			date.add(1, "day")
		}
		return result
	}

	_showDateTimePicker = (day, index, type) => {
		this.setState({
			isTimePickerVisible: true,
			pickedDay: day,
			pickedIndex: index,
			pickedType: type
		})
	}

	_hideDateTimePicker = () => this.setState({ isTimePickerVisible: false })

	_handleDatePicked = date => {
		const hour = moment(date).format("HH")
		const minutes = moment(date).format("mm")
		const { pickedDay, pickedIndex, pickedType } = this.state
		let newDay = [...this.state.daysWithHours[pickedDay]]
		newDay[pickedIndex] = {
			...newDay[pickedIndex],
			[`${pickedType}_hour`]: hour,
			[`${pickedType}_minutes`]: minutes
		}

		this.setState({
			daysWithHours: { ...this.state.daysWithHours, [pickedDay]: newDay }
		})

		this._hideDateTimePicker()
	}

	_addHours = day => {
		let newDay = [...this.state.daysWithHours[day], DEFAULT_HOURS]
		this.setState({
			daysWithHours: { ...this.state.daysWithHours, [day]: newDay }
		})
	}
	_renderHours = day => {
		return this.state.daysWithHours[day].map((hour, index) => {
			return (
				<View key={`hour${index}`} style={styles.hoursRow}>
					<TouchableOpacity
						onPress={() => {
							this._showDateTimePicker(day, index, "from")
						}}
						style={styles.hour}
					>
						<Text>
							{hour.from_hour.toString().padStart(2, "0")}:
							{hour.from_minutes.toString().padStart(2, "0")}
						</Text>
					</TouchableOpacity>
					<TouchableOpacity
						onPress={() => {
							this._showDateTimePicker(day, index, "to")
						}}
						style={styles.hour}
					>
						<Text>
							{hour.to_hour.toString().padStart(2, "0")}:
							{hour.to_minutes.toString().padStart(2, "0")}
						</Text>
					</TouchableOpacity>
				</View>
			)
		})
	}
	_renderDays = () => {
		return Object.keys(this.state.daysWithHours).map((day, index) => {
			let style = {}
			if (index == 0) {
				style = { marginTop: 0 }
			}
			return (
				<View key={`day${index}`} style={{ ...styles.day, ...style }}>
					<Text style={styles.dayTitle}>{day}</Text>
					{this._renderHours(day)}
					<TouchableOpacity
						style={styles.addButton}
						onPress={() => this._addHours(day)}
					>
						<Text style={styles.addButtonText}>
							{strings("teacher.work_days.add")}
						</Text>
					</TouchableOpacity>
				</View>
			)
		})
	}

	componentDidUpdate() {
		const error = this.props.dispatch(popLatestError(API_ERROR))
		if (error) {
			Alert.alert(strings("errors.title"), errors(error))
		}
	}

	save = async () => {
		console.log(this.state.daysWithHours)
		const resp = await this.props.dispatch(
			fetchOrError("/teacher/work_days", {
				method: "POST",
				body: JSON.stringify(this.state.daysWithHours)
			})
		)
		if (resp) {
			Alert.alert(strings("teacher.work_days.success"))
			this.props.navigation.goBack()
		}
	}

	render() {
		let title = strings("teacher.work_days.title")
		if (this.state.fromSchedule) {
			title = strings("teacher.work_days.title_from_schedule")
		}
		return (
			<Fragment>
				<ScrollView>
					<View style={styles.container}>
						<PageTitle
							style={styles.title}
							title={title}
							leftSide={
								<TouchableOpacity
									onPress={() => {
										this.props.navigation.dispatch(
											NavigationActions.back()
										)
									}}
									style={styles.closeButton}
								>
									<Icon
										name="ios-close"
										type="ionicon"
										size={36}
									/>
								</TouchableOpacity>
							}
						/>
						<View style={styles.days}>{this._renderDays()}</View>
						<DateTimePicker
							isVisible={this.state.isTimePickerVisible}
							mode={"time"}
							onConfirm={this._handleDatePicked.bind(this)}
							onCancel={this._hideDateTimePicker}
							is24Hour={true}
						/>
					</View>
				</ScrollView>
				<TouchableHighlight
					underlayColor="#ffffff00"
					onPress={this.save.bind(this)}
					style={fullButton}
				>
					<Text style={styles.buttonText}>
						{strings("teacher.work_days.save")}
					</Text>
				</TouchableHighlight>
			</Fragment>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		marginLeft: MAIN_PADDING,
		marginRight: MAIN_PADDING,
		marginTop: 20
	},
	title: {
		marginTop: 4
	},
	days: {
		flex: 1
	},
	day: {
		marginTop: 20
	},
	dayTitle: {
		fontWeight: "bold",
		marginLeft: 8,
		alignSelf: "center"
	},
	row: {
		flexDirection: "row",
		alignItems: "flex-start",
		justifyContent: "flex-start"
	},
	hoursRow: {
		flexDirection: "row",
		justifyContent: "center",
		marginTop: 12
	},
	hour: {
		width: 100,
		alignItems: "center",
		paddingVertical: 8,
		backgroundColor: "#f8f8f8",
		marginRight: 12
	},
	inputStyle: {},
	inputContainer: {
		maxWidth: 150
	},
	addButton: {
		marginTop: 8,
		alignSelf: "center",
		alignItems: "center"
	},
	addButtonText: {
		color: colors.blue,
		fontWeight: "bold"
	},
	closeButton: {
		marginTop: Platform.select({ ios: -6, android: -12 })
	},
	buttonText: {
		color: "#fff",
		fontWeight: "bold"
	}
})
function mapStateToProps(state) {
	return {
		errors: state.errors,
		fetchService: state.fetchService
	}
}
export default connect(mapStateToProps)(WorkDays)
