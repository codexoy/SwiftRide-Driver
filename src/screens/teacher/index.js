import React from "react"
import {
	createBottomTabNavigator,
	createStackNavigator
} from "react-navigation"
import Home from "./Home"
import Notifications from "./Notifications"
import ChooseDate from "./ChooseDate"
import NewLesson from "./NewLesson"
import Students from "./Students"
import NewStudent from "./NewStudent"
import Schedule from "./Schedule"
import { strings } from "../../i18n"
import { Icon } from "react-native-elements"

export default createBottomTabNavigator(
	{
		Home: Home,
		Notifications: Notifications,
		Add: {
			screen: createStackNavigator(
				{ ChooseDate: ChooseDate, NewLesson: NewLesson },
				{
					initialRouteName: "ChooseDate",
					headerMode: "none",
					navigationOptions: {
						headerVisible: false
					}
				}
			),
			navigationOptions: {
				title: "add",
				tabBarLabel: strings("tabs.add"),
				tabBarAccessibilityLabel: strings("tabs.add"),
				tabBarTestID: "NewLessonTab"
			}
		},
		Schedule: Schedule,
		Students: {
			screen: createStackNavigator(
				{ Students: Students, NewStudent: NewStudent },
				{
					mode: "modal",
					initialRouteName: "Students",
					headerMode: "none",
					navigationOptions: {
						headerVisible: false
					}
				}
			),
			navigationOptions: {
				title: "students",
				tabBarLabel: strings("tabs.students"),
				tabBarAccessibilityLabel: strings("tabs.students"),
				tabBarTestID: "StudentsTab"
			}
		}
	},
	{
		defaultNavigationOptions: ({ navigation }) => ({
			tabBarIcon: ({ focused, horizontal, tintColor }) => {
				const { routeName } = navigation.state
				let iconName
				let iconType = "material"
				if (routeName === "Home") {
					iconName = "home"
				} else if (routeName === "Notifications") {
					iconName = "notifications"
				} else if (routeName === "Add") {
					iconName = "add-circle"
				} else if (routeName === "Students") {
					iconName = "people"
				} else if (routeName == "Schedule") {
					iconName = "date-range"
				}

				// You can return any component that you like here!
				return (
					<Icon
						name={iconName}
						size={25}
						color={tintColor}
						type={iconType}
					/>
				)
			}
		}),
		tabBarOptions: {
			activeTintColor: "rgb(12,116,244)",
			inactiveTintColor: "black",
			showLabel: false
		}
	}
)
