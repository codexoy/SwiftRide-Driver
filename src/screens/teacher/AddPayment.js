import { createStackNavigator } from "react-navigation"
import AddPaymentChooseStudent from "./AddPaymentChooseStudent"
import AddPaymentChooseAmount from "./AddPaymentChooseAmount"
export default createStackNavigator(
	{ First: AddPaymentChooseStudent, Second: AddPaymentChooseAmount },
	{
		initialRouteKey: "First",
		headerMode: "none",
		navigationOptions: {
			headerVisible: false
		}
	}
)
