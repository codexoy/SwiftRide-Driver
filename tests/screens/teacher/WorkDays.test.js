import "react-native"
import React from "react"

import renderer from "react-test-renderer"
import FetchService from "../../../src/services/Fetch"
import { WorkDays } from "../../../src/screens/teacher/WorkDays"

const fetchService = new FetchService()
jest.useFakeTimers()
// currently it will only render loading,
// until we fix the async tests
describe("WorkDays", () => {
	test("view renders correctly", () => {
		const tree = renderer.create(
			<WorkDays
				fetchService={fetchService}
				navigation={{ getParam: jest.fn() }}
			/>
		)

		expect(tree.toJSON()).toMatchSnapshot()
	})
})
