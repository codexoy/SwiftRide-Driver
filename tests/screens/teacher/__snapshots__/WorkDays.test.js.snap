// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkDays view renders correctly 1`] = `
<Fragment>
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 26,
        "marginRight": 26,
        "marginTop": 20,
      }
    }
  >
    <PageTitle
      leftSide={
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Object {
              "marginTop": -6,
            }
          }
        >
          <Themed.Icon
            name="ios-close"
            size={36}
            type="ionicon"
          />
        </TouchableOpacity>
      }
      style={
        Object {
          "marginTop": 4,
        }
      }
      title="שעות עבודה קבועות"
    />
    <ScrollViewMock
      style={
        Object {
          "marginBottom": 68,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          key="day0"
          style={
            Object {
              "marginTop": 0,
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "fontWeight": "bold",
                "marginLeft": 8,
              }
            }
          >
            ראשון
             
          </Text>
          <View
            key="hour0"
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
                "marginTop": 12,
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#f8f8f8",
                  "marginRight": 12,
                  "paddingVertical": 8,
                  "width": 100,
                }
              }
            >
              <Text>
                16
                :
                00
              </Text>
            </TouchableOpacity>
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#f8f8f8",
                  "marginRight": 12,
                  "paddingVertical": 8,
                  "width": 100,
                }
              }
            >
              <Text>
                20
                :
                00
              </Text>
            </TouchableOpacity>
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <Text
                style={
                  Object {
                    "color": "rgb(12,116,244)",
                    "marginTop": 6,
                  }
                }
              >
                הסר
              </Text>
            </TouchableOpacity>
          </View>
          <View
            key="hour1"
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
                "marginTop": 12,
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#f8f8f8",
                  "marginRight": 12,
                  "paddingVertical": 8,
                  "width": 100,
                }
              }
            >
              <Text>
                08
                :
                00
              </Text>
            </TouchableOpacity>
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#f8f8f8",
                  "marginRight": 12,
                  "paddingVertical": 8,
                  "width": 100,
                }
              }
            >
              <Text>
                14
                :
                00
              </Text>
            </TouchableOpacity>
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <Text
                style={
                  Object {
                    "color": "rgb(12,116,244)",
                    "marginTop": 6,
                  }
                }
              >
                הסר
              </Text>
            </TouchableOpacity>
          </View>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "marginTop": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "rgb(12,116,244)",
                  "fontWeight": "bold",
                }
              }
            >
              הוסף שעות
            </Text>
          </TouchableOpacity>
        </View>
        <View
          key="day1"
          style={
            Object {
              "marginTop": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "fontWeight": "bold",
                "marginLeft": 8,
              }
            }
          >
            שני
             
          </Text>
          <View
            key="hour0"
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
                "marginTop": 12,
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#f8f8f8",
                  "marginRight": 12,
                  "paddingVertical": 8,
                  "width": 100,
                }
              }
            >
              <Text>
                08
                :
                00
              </Text>
            </TouchableOpacity>
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#f8f8f8",
                  "marginRight": 12,
                  "paddingVertical": 8,
                  "width": 100,
                }
              }
            >
              <Text>
                14
                :
                00
              </Text>
            </TouchableOpacity>
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <Text
                style={
                  Object {
                    "color": "rgb(12,116,244)",
                    "marginTop": 6,
                  }
                }
              >
                הסר
              </Text>
            </TouchableOpacity>
          </View>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "marginTop": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "rgb(12,116,244)",
                  "fontWeight": "bold",
                }
              }
            >
              הוסף שעות
            </Text>
          </TouchableOpacity>
        </View>
        <View
          key="day2"
          style={
            Object {
              "marginTop": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "fontWeight": "bold",
                "marginLeft": 8,
              }
            }
          >
            שלישי
             
          </Text>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "marginTop": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "rgb(12,116,244)",
                  "fontWeight": "bold",
                }
              }
            >
              הוסף שעות
            </Text>
          </TouchableOpacity>
        </View>
        <View
          key="day3"
          style={
            Object {
              "marginTop": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "fontWeight": "bold",
                "marginLeft": 8,
              }
            }
          >
            רביעי
             
          </Text>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "marginTop": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "rgb(12,116,244)",
                  "fontWeight": "bold",
                }
              }
            >
              הוסף שעות
            </Text>
          </TouchableOpacity>
        </View>
        <View
          key="day4"
          style={
            Object {
              "marginTop": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "fontWeight": "bold",
                "marginLeft": 8,
              }
            }
          >
            חמישי
             
          </Text>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "marginTop": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "rgb(12,116,244)",
                  "fontWeight": "bold",
                }
              }
            >
              הוסף שעות
            </Text>
          </TouchableOpacity>
        </View>
        <View
          key="day5"
          style={
            Object {
              "marginTop": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "fontWeight": "bold",
                "marginLeft": 8,
              }
            }
          >
            שישי
             
          </Text>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "marginTop": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "rgb(12,116,244)",
                  "fontWeight": "bold",
                }
              }
            >
              הוסף שעות
            </Text>
          </TouchableOpacity>
        </View>
        <View
          key="day6"
          style={
            Object {
              "marginTop": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "fontWeight": "bold",
                "marginLeft": 8,
              }
            }
          >
            שבת
             
          </Text>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "marginTop": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "rgb(12,116,244)",
                  "fontWeight": "bold",
                }
              }
            >
              הוסף שעות
            </Text>
          </TouchableOpacity>
        </View>
      </View>
      <CustomDatePickerIOS
        cancelTextIOS="Cancel"
        confirmTextIOS="Confirm"
        date={2019-06-18T09:00:00.195Z}
        hideTitleContainerIOS={false}
        is24Hour={true}
        isVisible={false}
        mode="time"
        neverDisableConfirmIOS={false}
        onCancel={[Function]}
        onConfirm={[Function]}
        onDateChange={[Function]}
        onHideAfterConfirm={[Function]}
        reactNativeModalPropsIOS={Object {}}
        titleIOS="Pick a date"
      />
    </ScrollViewMock>
  </View>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(12,116,244)",
        "borderRadius": 0,
        "bottom": 0,
        "elevation": 8,
        "height": 56,
        "justifyContent": "center",
        "position": "absolute",
        "shadowColor": "rgb(12,116,244)",
        "shadowOffset": Object {
          "height": 8,
          "width": 0,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 16,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontWeight": "bold",
        }
      }
    >
      שמור
    </Text>
  </TouchableOpacity>
</Fragment>
`;
